src/Product/ProductInfo.cs

```diff
@@ -8,44 +8,61 @@ namespace Azure.DataApiBuilder.Product;
 
 public static class ProductInfo
 {
-    public const string DEFAULT_VERSION = "0.0.0";
     public const string DAB_APP_NAME_ENV = "DAB_APP_NAME_ENV";
-    public static readonly string DEFAULT_APP_NAME = $"dab_oss_{ProductInfo.GetProductVersion()}";
-    public static readonly string ROLE_NAME = "DataApiBuilder";
+    public static readonly string DAB_USER_AGENT = $"dab_oss_{GetMajorMinorPatchVersion()}";
+    public static readonly string DEFAULT_APP_NAME = $"dab_oss_{GetProductVersion()}";
+    public static readonly string CLOUD_ROLE_NAME = "DataApiBuilder";
 
     /// <summary>
     /// Reads the product version from the executing assembly's file version information.
+    /// Includes commit hash.
     /// </summary>
+    /// <param name="includeCommitHash">True by default: the user agent string will include the commit hash.</param>
     /// <returns>Product version if not null, default version 0.0.0 otherwise.</returns>
-    public static string GetProductVersion()
+    public static string GetProductVersion(bool includeCommitHash = true)
```

Comment:
Why do we need separate logic for GetProductVersion() v/s GetMajorMinorPatchVersion()? 

I dont think we ever desire the commit hash to be returned anywhere. 
We dont need includeCommitHash argument.

Reply:
consolidated to GetProductVersion and removed any usage of commit hash.

Reply:
Revised: Dab health endpoint does not reference commit hash. Commit hash only referenced in CLI due to how we use CommandLineParser dependency. #1456 tracks the fix. But I still consolidated GetProductVersion() v/s GetMajorMinorPatchVersion() as you suggested here.